### ISSUE TITLE
Treat SIXELCH(0x01) as a space character when it is copied to clipboard

### ISSUE THREAD

**Issue by saitoha** [mintty/mintty#587]
When we select an area which includes SIXEL images with mouse and copy it, some internal characters SIXELCH(\x01) are put into the clipboard.
I think these should be replaced with space characters.

![copy](https://cloud.githubusercontent.com/assets/1162739/18417263/95c73f84-7865-11e6-8c1d-8fef13481688.gif)

patch: https://github.com/saitoha/mintty-sixel/commit/959418bbdc8d683f27d0fcf00b2ff7c6596baa89

**Comment by mintty** [comment]
See https://github.com/saitoha/mintty-sixel/commit/959418bbdc8d683f27d0fcf00b2ff7c6596baa89#commitcomment-18977353

Is this in any way related to the weird rendering of an image while under a selection (same display as flickering for an instance before SIXEL images are being refreshed):
![sixel-select](https://cloud.githubusercontent.com/assets/12740416/18418552/f439559c-7848-11e6-9af6-41037025638b.png)

**Comment by mintty** [comment]
I think I'd rather go with SIXELCH = 0xFFFC and change that. As 0xFFFC looks like SPACE in many fonts, one could also consider 0xFFFD instead, but that would not be correct according to its Unicode description.

Any idea about my screenshot above? Can you reproduce it? (Probably a separate issue...)

**Comment by saitoha** [comment]
Hmmm... Here is screenshots on my environment (32bit Windows 7 sp 1)

- Lucida Console (Default font)
<img width="619" alt="lucida_console" src="https://cloud.githubusercontent.com/assets/1162739/18443091/5a3e274a-794e-11e6-9efd-d4141e1cf053.png">

- Consolas
<img width="577" alt="consolas" src="https://cloud.githubusercontent.com/assets/1162739/18443164/99f27828-794e-11e6-9b91-9ff0b70a3dd1.png">

**Comment by saitoha** [comment]
Does this workaround work for you?

```
diff --git a/src/term.c b/src/term.c
index 6dc8f68..bb43d24 100644
--- a/src/term.c
+++ b/src/term.c
@@ -1162,7 +1162,7 @@ term_paint(void)
         !termchars_equal_override(&dispchars[j], d, tchar, tattr);
       dirty_run |= do_copy;
 
-      text[textlen++] = tchar;
+      text[textlen++] = tchar == SIXELCH ? ' ' : tchar;
       if (!has_rtl)
         has_rtl = is_rtl_class(tbc);
 
diff --git a/src/term.h b/src/term.h
index 4609a4d..2b9f82c 100644
--- a/src/term.h
+++ b/src/term.h
@@ -65,7 +65,7 @@ typedef enum {
  * SIXELCH is a special character which represents a part of SIXEL graphics.
  */
 enum { UCSWIDE = 0,
-       SIXELCH = 0xFFFC
+       SIXELCH = 1
 };
 
 /* Three character attribute types:
```

I prefer SIXELCH is a ascii control character because it is easily distinguished with display characters.

**Comment by mintty** [comment]
Thanks, the patch fixes the rendering glitch. I would rather make it more explicit:
```
      if (tchar == SIXELCH)
        // indicate visually that the image will not be copied
        text[textlen++] = 0xFFFD;
      else
        text[textlen++] = tchar;
```
And I still think that SIXELCH = 0xFFFC (which is not really a display character) is a good choice, also for pasting, and that a control char code might cause further confusion in an unforeseen context, maybe.

**Comment by mintty** [comment]
Apparently the display replacement I intended for selection also appears for a flickering moment if the screen is scrolled. This is unpleasant, so I will change that to pure space again.

**Comment by saitoha** [comment]
I'm glad if SIXEL characters are represented as space characters in the clipboard, not only in the display.

![copy](https://cloud.githubusercontent.com/assets/1162739/18793803/6f5cf1c2-81f7-11e6-80d0-3b89392c1d97.gif)

Wired characters "???????..."  are not needed for me...

**Comment by mintty** [comment]
OK, I'll change it, at least by default. Maybe I'll make it configurable.

**Comment by saitoha** [comment]
Confirmed. Thanks!

**Comment by mintty** [comment]
Released 2.6.2.
